be/src/column/array_column.cpp

```diff
@@ -438,7 +438,8 @@ void ArrayColumn::compare_column(const Column& rhs_column, std::vector<int8_t>*
     size_t rows = size();
     output->resize(rows);
     for (size_t i = 0; i < rows; i++) {
-        (*output)[i] = static_cast<int8_t>(compare_at(i, i, rhs_column, 1));
+        int res = compare_at(i, i, rhs_column, 1);
+        (*output)[i] = (res > 0) - (res < 0);
```

Comment:
why not change the res from `memcmp `, but here?

Reply:
Can you add a comment here to explain why this code is written like this.